README.md

```diff
@@ -19,113 +19,122 @@ This library includes the following modules.
-
-// an example with ScalaCheck using the Arbitrary typeclass
-// this example is a bit more practical, since there's no typeclass instantiation built into Scalacheck, unlike Cats
+import cats._
+import cats.instances.all._ // implicit instances for Semigroup[Int], etc.
+val sg: Semigroup[Outer] = implicitly[Semigroup[Outer]]
+sg.combine(Outer(Inner(1, "hello, ")), Outer(Inner(100, "world!")))
+// = Outer(Inner(101,hello, world!))
+
+// an example with ScalaCheck Arbitrary
 import magnolify.scalacheck.auto._
-import org.scalacheck.Arbitrary
-implicit val arbString: Arbitrary[String] = ??? // could be Arbitrary.arbitrary[String]
-val arbitraryCaseClass: Arbitrary[MyCaseClass] = implicitly[Arbitrary[MyCaseClass]]
-
-// an example with Guava using the Funnel typeclass
-import magnolify.guava.auto._
-import com.google.common.hash.Funnel
-implicit val funnelString: Funnel[String] = ??? // could be Funnels.stringFunnel(charset)
-val implicitly[Funnel[MyCaseClass]]
+import org.scalacheck._ // implicit instances for Arbitrary[Int], etc.
+val arb: Arbitrary[Outer] = implicitly[Arbitrary[Outer]]
+arb.arbitrary.sample
+// = Some(Outer(Inter(12345, abcde)))
+
+// an example with Guava Funnel
+import magnolify.guava.auto._ // includes implicit instances for Funnel[Int], etc.
+import com.google.common.hash._
+val fnl: Funnel[Outer] = implicitly[Funnel[Outer]]
+BloomFilter.create[Outer](fnl, 1000) // a BloomFilter that works with Outer
 ```
 
-Semi-automatic derivation happens explicitly by calling the derivation yourself instead of summoning an implicit.
+Semi-automatic derivation needs to be called explicitly.
+
 ```scala
 import magnolify.cats.semiauto._
-EqDerivation[MyCaseClass]
-HashDerivation[MyCaseClass]
-SemigroupDerivation[MyCaseClass]
-MonoidDerivation[MyCaseClass]
-GroupDerivation[MyCaseClass]
+import cats._
+import cats.instances.all._
+EqDerivation[Outer]
+HashDerivation[Outer]
+SemigroupDerivation[Outer]
+MonoidDerivation[Outer]
+// this fails due to missing `Group[String]` instance
+// GroupDerivation[Outer]
 
 import magnolify.scalacheck.semiauto._
-ArbitraryDerivation[MyCaseClass]
-CogenDerivation[MyCaseClass]
+import org.scalacheck._
+ArbitraryDerivation[Outer]
+CogenDerivation[Outer]
 
 import magnolify.guava.semiauto._
-FunnelDerivation[MyCaseClass]
+FunnelDerivation[Outer]
 ```
 
-Conversion between a case class instance and an instance of another type (e.g. GenericRecord) containing the same data must be called explicitly/semi-automatically, specifying the case class to be used for conversion, rather than deriving entirely with implicits.
+Typeclasses for data type conversion must be called explicitly.
 
 ```scala
-// case classes used in examples
-case class MyCaseClass(field: NestedCaseClass) 
-case class NestedCaseClass(innerField: String) // works with other non-String elements, this is just an example
+import java.net.URI
+case class Inner(long: Long, str: String, uri: URI)
```

Comment:
`Long` instead of `Int` since Datastore & TF.Example only support the former. `URI` to illustrate custom types.